includes/amp-helper-functions.php

```diff
@@ -579,23 +579,46 @@ function amp_get_schemaorg_metadata() {
 		),
 	);
 
-	/**
-	 * Filters the site icon used in AMP responses.
-	 *
-	 * In general the `get_site_icon_url` filter should be used instead.
-	 *
-	 * @since 0.3
-	 * @todo Why is the size set to 32px?
-	 *
-	 * @param string $site_icon_url
-	 */
-	$site_icon_url = apply_filters( 'amp_site_icon_url', get_site_icon_url( AMP_Post_Template::SITE_ICON_SIZE ) );
-	if ( $site_icon_url ) {
-		$metadata['publisher']['logo'] = array(
-			'@type'  => 'ImageObject',
-			'url'    => $site_icon_url,
-			'height' => AMP_Post_Template::SITE_ICON_SIZE,
-			'width'  => AMP_Post_Template::SITE_ICON_SIZE,
+	$custom_logo_id = get_theme_mod( 'custom_logo' );
+	if ( $custom_logo_id && 'attachment' === get_post_type( $custom_logo_id ) ) {
+		$custom_logo_img = wp_get_attachment_image_src( $custom_logo_id, 'full', false );
```

Comment:
**Documentation**

We might want to mention the [logo guidelines](https://developers.google.com/search/docs/data-types/article#logo-guidelines) in our documentation. Otherwise, it's unlikely that the custom logo image will fit inside the required 60 x 600px rectangle.

The suggested custom logo sizes are:

[Twenty Fifteen](https://github.com/WordPress/wordpress-develop/blob/805b2761a9e17ddea2c1c397c1ba5d4d792d4fa0/src/wp-content/themes/twentyfifteen/functions.php#L131): 248 x 248
[Twenty Sixteen](https://github.com/WordPress/wordpress-develop/blob/805b2761a9e17ddea2c1c397c1ba5d4d792d4fa0/src/wp-content/themes/twentysixteen/functions.php#L74): 240 x 240
[Twenty Seventeen](https://github.com/WordPress/wordpress-develop/blob/805b2761a9e17ddea2c1c397c1ba5d4d792d4fa0/src/wp-content/themes/twentyseventeen/functions.php#L101): 250 x 250

Reply:
Also, we might consider falling back to the Site Icon if the Custom Logo's dimensions are too big.

The plugin currently outputs the 32 x 32px Site Icon. Strangely, it doesn't cause an error in the Google Structured Data Testing Tool, even though it doesn't follow the [logo guidelines](https://developers.google.com/search/docs/data-types/article#logo-guidelines):

[Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool/u/0/?hl=es#url=https%3A%2F%2Fmake.xwp.co%2F2018%2F02%2F28%2Fmy-thoughts-on-leadership%2Famp%2F)
<img width="1408" alt="structured-data-testing-tool-current" src="https://user-images.githubusercontent.com/4063887/39971830-4e69ee14-56c8-11e8-92bb-b311baa9bdd8.png">


With this PR applied, there's an error in that tool if the image size is too big:
<img width="1418" alt="structured-data-1144" src="https://user-images.githubusercontent.com/4063887/39971850-cba37c56-56c8-11e8-9fca-f2cf896555b0.png">